cynghanedd.py

```diff
@@ -205,14 +216,57 @@ def choose_word(solutions, word):
 
     words = []
 
-    while solution and solution[0]:
+    words, solution = complete_cyng(words, skeleton, solution)
+
+    return words, skeleton, solution
+
+
+def back_cynghanedd(old_words, skeleton):
+    solution = search(skeleton)
+
+    print(old_words[:-1])
```

Comment:
You can remove this line, although it likes you were experimenting a bit with how to slice pieces off of `old_words`, and I think this expression is indeed what you want! `old_words[:-1]` is everything in `old_words` except for its final element.